src/gabarit_sup.erl

```diff
@@ -2,34 +2,36 @@
 %% @doc gabarit top level supervisor.
 %% @end
 %%%-------------------------------------------------------------------
-
 -module(gabarit_sup).
-
 -behaviour(supervisor).
-
 -export([start_link/0]).
-
 -export([init/1]).
-
 -define(SERVER, ?MODULE).
 
 start_link() ->
     supervisor:start_link({local, ?SERVER}, ?MODULE, []).
 
-%% sup_flags() = #{strategy => strategy(),         % optional
-%%                 intensity => non_neg_integer(), % optional
-%%                 period => pos_integer()}        % optional
-%% child_spec() = #{id => child_id(),       % mandatory
-%%                  start => mfargs(),      % mandatory
-%%                  restart => restart(),   % optional
-%%                  shutdown => shutdown(), % optional
-%%                  type => worker(),       % optional
-%%                  modules => modules()}   % optional
 init([]) ->
     SupFlags = #{strategy => one_for_all,
                  intensity => 0,
                  period => 1},
-    ChildSpecs = [],
-    {ok, {SupFlags, ChildSpecs}}.
+    %% TODO: which option is better for the application
+    StoreOpts = case application:get_env(gabarit, persistence, none) of
```

Comment:
I usually prefer to have a specific function to deal with options somewhere else than in the `init/1` function.